docs/examples/metrics/measurement-processors/README.md

```diff
@@ -0,0 +1,245 @@
+2. **Modify and pass**: `next_processor(modified_measurement)`
+3. **Drop measurement**: Simply don't call `next_processor`
+4. **Split into multiple**: Call `next_processor` multiple times
+
+## Usage
+
+### Basic Setup
+
+```python
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics._internal.measurement_processor import (
+    BaggageMeasurementProcessor,
+    StaticAttributeMeasurementProcessor,
+    ValueRangeMeasurementProcessor,
+)
+
+# Create measurement processors
+processors = [
+    BaggageMeasurementProcessor(),  # Add baggage as attributes
+    StaticAttributeMeasurementProcessor({"env": "prod"}),  # Add static attributes
+    ValueRangeMeasurementProcessor(min_value=0),  # Drop negative values
+]
+
+# Configure MeterProvider with processors
+meter_provider = MeterProvider(
+    measurement_processors=processors,
+    # ... other configuration
+)
+```
+
+### Built-in Processors
+
+#### 1. BaggageMeasurementProcessor
+
+Extracts values from OpenTelemetry Baggage and adds them as measurement attributes, enabling end-to-end telemetry correlation.
+
+```python
+# Add all baggage as attributes with "baggage." prefix
+processor = BaggageMeasurementProcessor()
+
+# Add only specific baggage keys
+processor = BaggageMeasurementProcessor(baggage_keys=["user.id", "trace.id"])
+```
+
+#### 2. StaticAttributeMeasurementProcessor
+
+Adds static attributes to all measurements.
+
+```python
+processor = StaticAttributeMeasurementProcessor({
+    "environment": "production",
+    "service": "api-server",
+    "version": "1.0.0"
+})
+```
+
+#### 3. AttributeFilterMeasurementProcessor
+
+Removes specific attributes from measurements (useful for removing sensitive data).
+
+```python
+processor = AttributeFilterMeasurementProcessor([
+    "password", "secret", "auth_token"
+])
+```
+
+#### 4. ValueRangeMeasurementProcessor
+
+Drops measurements outside a specified value range.
+
+```python
+# Drop negative values and values over 1000
+processor = ValueRangeMeasurementProcessor(min_value=0, max_value=1000)
+
+# Only minimum limit
+processor = ValueRangeMeasurementProcessor(min_value=0)
+
+# Only maximum limit  
+processor = ValueRangeMeasurementProcessor(max_value=100)
+```
+
+### Custom Processors
+
+Create custom processors by implementing the `MeasurementProcessor` interface:
+
+```python
+from opentelemetry.sdk.metrics._internal.measurement_processor import MeasurementProcessor
+from opentelemetry.sdk.metrics._internal.measurement import Measurement
+from dataclasses import replace
+from typing import Callable
+
+class CustomMeasurementProcessor(MeasurementProcessor):
+    def process(
+        self, 
+        measurement: Measurement, 
+        next_processor: Callable[[Measurement], None]
+    ) -> None:
+        # Example: Add timestamp attribute
+        new_attributes = dict(measurement.attributes or {})
+        new_attributes["processed_at"] = str(int(time.time()))
```

Comment:
I don't think we should show example of adding timestamp as a Metric attribute - is there any valid use-case for it?